ops/testing.py

```diff
@@ -135,6 +135,47 @@ class ExecResult:
 ExecHandler = Callable[[ExecArgs], Union[None, ExecResult]]
 
 
+@dataclasses.dataclass(frozen=True)
+class ActionOutput:
+    """Contains the logs and results from a :meth:`Harness.run_action` call."""
+
+    logs: List[str]
+    """Messages generated by the Charm using :meth:`ops.ActionEvent.log`."""
+
+    results: Dict[str, str]
```

Comment:
```suggestion
    results: Dict[str, Any]
```

Reply:
Good catch, thanks! It's strange that pyright didn't complain about the `action_set` line that does a dictionary update of a dict[str, Any] into a dict[str, str].